spaceship/lib/spaceship/portal/certificate.rb

```diff
@@ -286,9 +332,17 @@ def create!(csr: nil, bundle_id: nil)
 
           # look up the app_id by the bundle_id
           if bundle_id
-            app = Spaceship::App.find(bundle_id)
+            app = Spaceship::App.find_by_platform(bundle_id)
             raise "Could not find app with bundle id '#{bundle_id}'" unless app
-            app_id = app.app_id
+            app_id = if app.app_id
+                       app.app_id
+                     elsif app.website_push_id
+                       app.website_push_id
+                     elsif app.pass_type_id
+                       app.pass_type_id
+                     elsif app.omc_id
+                       app.omc_id
+                     end
```

Comment:
Might be more simply stateable as:

```ruby
app_id = app.app_id || app.website_push_id || app.pass_type_id || app.omc_id
```

please double-check my logic if you like that approach :+1: